common/helpers/src/avro.rs

```diff
@@ -0,0 +1,235 @@
+/// If schema is malformed or invalid, returns an error.
+/// # Arguments
+/// * `serialized_schemas` - list of serialized Avro schema to be converted
+/// # Returns
+/// * `Result<Vec<Schema>, AvroError>` - structured Avro schema
+/// # Examples
+/// ```
+/// use avro::*;
+/// let raw_schema = r#"{"type": "record", "name": "User", "fields": [{"name": "name", "type": "string"}, {"name": "favorite_number", "type": "int"}]}"#;
+/// let serialized_schema = fingerprint_raw_schema(raw_schema);
+/// assert!(serialized_schema.is_ok());
+/// let schema = schema_result.unwrap().1;
+/// let translated_schema = translate_schemas(&vec![schema]);
+/// assert!(translated_schema.is_ok());
+/// ```
+pub fn translate_schemas(serialized_schemas: Vec<Vec<u8>>) -> Result<Vec<Schema>, AvroError> {
+	let mut schemas = Vec::new();
+	for s in serialized_schemas {
+		schemas.push(translate_schema(s)?);
+	}
+	Ok(schemas)
+}
+
+/// Function to get the schema writer with default container as Vec<u8>
+/// # Examples
+/// ```
+/// use avro::*;
+/// let writer = get_schema_data_writer();
+/// ```
+pub fn get_schema_data_writer<'a>(schema: &'a Schema) -> Writer<'a, Vec<u8>> {
+	Writer::with_codec(schema, Vec::new(), Codec::Snappy)
+}
+
+/// Function to populate a given schema writer with schema data.
+/// # Arguments
+/// * `writer` - io Writer ( can be obtained from get_schema_data_writer() )
+/// * `records` - record to be written as HashMap<String, SchemaValue> where SchemaValue is common types
+/// # Examples
+/// ```
+/// use avro::*;
+/// let writer = get_schema_data_writer();
+/// let record = vec![("name".to_string(), SchemaValue::String("John".to_string())), ("favorite_number".to_string(), SchemaValue::Int(42))];
+/// populate_records(writer, record);
+/// ```
+pub fn populate_records(
+	writer: &mut Writer<Vec<u8>>,
+	records: &HashMap<String, SchemaValue>,
+) -> Result<(), AvroError> {
+	let mut record_list = Record::new(writer.schema()).unwrap();
+	for (field_name, field_value) in records.iter() {
+		record_list.put(field_name, field_value.clone());
+	}
+	writer.append(record_list)?;
+	Ok(())
+}
+
+/// Function to populate a given schema with data and return serialized record.
+/// # Arguments
+/// * `schema` - Avro schema
+/// * `record` - record to be written as HashMap<String, SchemaValue> where SchemaValue is common types
+/// # Examples
+/// ```
+/// use avro::*;
+/// let schema = r#"{"type": "record", "name": "User", "fields": [{"name": "name", "type": "string"}, {"name": "favorite_number", "type": "int"}]}"#;
+/// let record = vec![("name".to_string(), SchemaValue::String("John".to_string())), ("favorite_number".to_string(), SchemaValue::Int(42))];
+/// let schema_fingerprint = fingerprint_raw_schema(schema);
+/// assert!(schema_fingerprint.is_ok());
+/// let serialized_record = populate_schema_and_serialize(schema_fingerprint.unwrap().1, record);
+pub fn populate_schema_and_serialize(
```

Comment:
It seems like you can reuse already existing functions `get_schema_data_writer` and `populate_records`

Reply:
Good point, refactored this